nested-types.ipynb

```diff
@@ -0,0 +1,802 @@
+       "- storage_type: 'struct'\n",
+       "- length: 5\n",
+       "- offset: 0\n",
+       "- null_count: 1\n",
+       "- buffers[1]:\n",
+       "  - validity <bool[1 b] 11011000>\n",
+       "- dictionary: NULL\n",
+       "- children[2]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'string'\n",
+       "    - length: 5\n",
+       "    - offset: 0\n",
+       "    - null_count: 1\n",
+       "    - buffers[3]:\n",
+       "      - validity <bool[1 b] 10111000>\n",
+       "      - data_offset <int32[24 b] 0 3 3 3 7 11>\n",
+       "      - data <string[11 b] b'joemarkjane'>\n",
+       "    - dictionary: NULL\n",
+       "    - children[0]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'int8'\n",
+       "    - length: 5\n",
+       "    - offset: 0\n",
+       "    - null_count: 1\n",
+       "    - buffers[2]:\n",
+       "      - validity <bool[1 b] 11110000>\n",
+       "      - data <int8[5 b] 1 2 0 4 0>\n",
+       "    - dictionary: NULL\n",
+       "    - children[0]:"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "na.c_array_view(na.c_array(column_4))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Map\n",
+    "\n",
+    "Map type represents nested data where each value is a variable number of key-item pairs. Its physical representation is the same as a list of `{key, item}` structs.\n",
+    "\n",
+    "![image info](./diagrams/map-diagram.svg)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<pyarrow.lib.MapArray object at 0x108077dc0>\n",
+       "[\n",
+       "  keys:\n",
+       "  [\n",
+       "    \"Dark Knight\"\n",
+       "  ]\n",
+       "  values:\n",
+       "  [\n",
+       "    10\n",
+       "  ],\n",
+       "  keys:\n",
+       "  [\n",
+       "    \"Dark Knight\",\n",
+       "    \"Meet the Parents\",\n",
+       "    \"Superman\"\n",
+       "  ]\n",
+       "  values:\n",
+       "  [\n",
+       "    8,\n",
+       "    4,\n",
+       "    5\n",
+       "  ],\n",
+       "  null,\n",
+       "  keys:\n",
+       "  [\n",
+       "    \"Meet the Parents\",\n",
+       "    \"Superman\"\n",
+       "  ]\n",
+       "  values:\n",
+       "  [\n",
+       "    10,\n",
+       "    null\n",
+       "  ]\n",
+       "]"
+      ]
+     },
+     "execution_count": 34,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_6_data = [[{'key': 'Dark Knight', 'value': 10}],\n",
+    "                 [{'key': 'Dark Knight', 'value': 8}, {'key': 'Meet the Parents', 'value': 4}, {'key': 'Superman', 'value': 5}],\n",
+    "                 None,\n",
+    "                 [{'key': 'Meet the Parents', 'value': 10}, {'key': 'Superman', 'value': None}]]\n",
```

Comment:
```suggestion
    "column_6_data = [{'Dark Knight': 10},\n",
    "                 {'Dark Knight': 8, 'Meet the Parents': 4, 'Superman': 5},\n",
    "                 None,\n",
    "                 {'Meet the Parents': 10, 'Superman': None}]\n",
```

While they key: value lists are the correct physical representation, I think for the construction from python it makes more sense to start from the logical key: value dictionaries